context/app/routes_cells.py

```diff
@@ -15,11 +15,76 @@ def cells_ui():
     )
 
 
-@blueprint.route('/cells.json', methods=['POST'])
-def cells_api():
+@blueprint.route('/cells/datasets-selected-by-gene.json', methods=['POST'])
+def datasets_selected_by_gene():
+    # Refine what it means for a dataset to match a cells query:
+    # It’s probably not useful to include a dataset if only a single
+    # cell highly expresses the gene in question.
+    # Besides a gene, and an expression level, probably want to say
+    # the percent of the total cells in the assay that match.
+
+    gene_name = request.args.get('gene_name')
+    min_gene_expression = request.args.get('min_gene_expression')
+    min_cell_percentage = request.args.get('min_cell_percentage')
+
+    client = Client(current_app.config['CELLS_API_ENDPOINT'])
+
+    # WARNING: This does not work with the current version of the client.
+    dataset_set = client.select_datasets(
+        where='gene',
+        has=[f'{gene_name} > {min_gene_expression}'],
+        min_cell_percentage=min_cell_percentage,
+        genomic_modality='rna'
+    )
+    return {
+        'results': list(dataset_set.get_list())
+    }
+
+
+@blueprint.route('/cells/cell-counts-for-datasets.json', methods=['POST'])
+def cell_counts_for_datasets():
+    # Select a set of datasets where cells express a given gene, and for each dataset,
+    # give the number of matching cells, and the total number of cells.
+
+    uuids = request.args.getlist('uuid')
+    gene_name = request.args.get('gene_name')
+    min_gene_expression = request.args.get('min_gene_expression')
+
     client = Client(current_app.config['CELLS_API_ENDPOINT'])
-    datasets = client.select_datasets().get_list()[0:10]
```

Comment:
Once the client changes are made, we should be able to do something like:
datasets = client.select_datasets(
        where='gene',
        has=[f'{gene_name} > {min_gene_expression}'],
        genomic_modality='rna', min_cell_percentage = min_cell_percentage)
results = datasets.get_list(values_included=[f'{gene_name} > {min_gene_expression}'])
And that will give you, along with the UUIDs for each dataset, the percentage of cells that meet the criteria provided.

Reply:
Not quite clear what the change is: Would we get rid of `has=[f'{gene_name} > {min_gene_expression}']` and replace it with `values_included=[f'{gene_name} > {min_gene_expression}']`? (Not really a question... will be more clear when it's delivered. Thanks!)